### PR TITLE
Heads up

### PR DESCRIPTION
**Pull request by kylebisley**
Regularly makes it past 500 moves. Surprisingly good when alone on board. 

Adds functions:
heads_up
target_selection
buffet
shortest_option
path_to_tail
chasing_tail
print_look_from_object (in tests file)

Removes functions:
get_min_path_to_food

### CODE REVIEW

File: app/board.py [hunk 1]
```diff
@@ -209,4 +209,4 @@ def look_from_here(self, the_tile, j_data):
         if not tile_is_head:
             self.get_tile_at(head["x"], head["y"]).set_cost(head_cost)
 
-        return [pathable_tiles, blocking_tiles]
\ No newline at end of file
+        return [pathable_tiles, blocking_tiles]
```
Comment (kylebisley) [inline app/board.py:212]: Just adds a new line. I've my vs code default to this standard. It'll be added everytime I git add all so probably best to leave in.

File: app/parse_board.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 SMALLER_SNAKE_FUTURE_HEAD = 15
 OUR_HEAD = 10
 LARGER_SNAKE_FUTURE_HEAD = 99
-PATHABLE_TAIL = 25
+PATHABLE_TAIL = 24
```
Comment (kylebisley) [inline app/parse_board.py:10]: This marks where tails have been removed visually in the path_array. Was helpful with debugging so I left it in.

File: app/tests.py [hunk 3]
```diff
@@ -29,3 +29,45 @@ def print_look_from(board_object, converted_data):
             print str(y) + " ",
         print ""
     print ""
+
+
```
Comment (kylebisley) [inline app/tests.py:33]: Added for flexibility in printing the pathable/walls data. Not called in the snake at the moment.

File: app/tests.py [hunk 4]
```diff
@@ -29,3 +29,45 @@ def print_look_from(board_object, converted_data):
...
+        converted_data (dict): Converted JSON data
+    '''
+    head_tile = board_object.get_tile_at(converted_data["you"]["body"][0]["x"],
+                                         converted_data["you"]["body"][0]["y"])
+
+    look_board_width = converted_data["board"]["width"]
+    look_board_height = converted_data["board"]["height"]
+    look_board = [["-" for x in range(look_board_width)] for y in range(look_board_height)]
+    
+    for tile in tileLists[0]:
```
Comment (AGuyNamedTate) [inline app/tests.py:50]: What is tileLists[0]?

File: app/main.py [hunk 5]
```diff
@@ -193,6 +158,159 @@ def cardinal(converted_data, path):
...
+    possible_futures = []
+    for neighbour in neighbours:
+        possible_futures.append(board.look_from_here(neighbour,
+                                                     converted_data))
+    return possible_futures
+
+
+def target_selection(converted_data, board):
+    '''
+    Can eventually be call list for logic to deal with oh shit situations.
```
Comment (kylebisley) [inline app/main.py:191]: Or any agro logic.

File: app/main.py [hunk 6]
```diff
@@ -193,6 +158,159 @@ def cardinal(converted_data, path):
...
+    possible_futures = heads_up(converted_data, board)
+    possible_futures = chasing_tail(possible_futures, converted_data, board)
+    options = buffet(possible_futures, converted_data, board)
+
+    shortest_path = "Unassigned"
+
+    if len(options) > 0:
+        shortest_path = shortest_option(options)
+    # No safe food so path to tail
+    if (len(options) == 0) or (shortest_path == "Unassigned"):
```
Comment (kylebisley) [inline app/main.py]: can be cleaner

**Review by AGuyNamedTate — APPROVED** [review]

**Pull request merged**